packages/@pollyjs/adapter-xhr/src/index.js

```diff
@@ -36,11 +45,13 @@ export default class XHRAdapter extends Adapter {
       };
     };
 
-    global.XMLHttpRequest[IS_STUBBED] = true;
+    context.XMLHttpRequest[IS_STUBBED] = true;
```

Comment:
Weakmap could work here so we do mutate the objects we don't own

Reply:
Updated to use a `WeakSet`